### PR TITLE
libb64 1.2 (new formula)

### PR DESCRIPTION
**Pull request by schloegl**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/libb64.rb
Comment (Bo98) [inline Formula/libb64.rb]: Could we create something more useful here involving assertions? Like maybe create a file, feed that in, and see if the output is as expected?

Comment (schloegl) [reply Formula/libb64.rb]: For my other tools, I need only libb64. For that purpose starting base64 seems to be a good enough test.  However, feel free to add a more elaborate test, if you like.

Comment (Bo98) [reply Formula/libb64.rb]: Right, gotcha. That's fine.

I'll have a look later this week before merging. Thanks for your first pull request @schloegl!

Comment (zbeekman) [reply Formula/libb64.rb]: > For my other tools, I need only libb64. For that purpose starting base64 seems to be a good enough test. However, feel free to add a more elaborate test, if you like.

This definitely needs a better test. Homebrew maintainers typically have their hands pretty full, so we'll probably need your help @schloegl to get this going.

If you only need the library, you could write a small C program in the tetst (or similar) to ensure you can link against the library and/or install a simple test into the `pkg_share` directory during install and then compile & link in the test.

File: Formula/libb64.rb
Comment (zbeekman) [inline Formula/libb64.rb]: macOS provides a `base64` command that this will shadow. This formula should be `keg_only` or should only install the `base64` binary into libexec.

```suggestion
    libexec.install "base64/base64"
```

I'm not sure which one is appropriate. It looks like base64 encoding is provided within macOS frameworks:

```sh
otool -L /usr/bin/base64
/usr/bin/base64:
        /System/Library/PrivateFrameworks/CoreDaemon.framework/Versions/B/CoreDaemon (compatibility version 3.0.0, current version 3.0.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1252.250.1)
```

Should this be keg only if accepted?

**Review by zbeekman — CHANGES_REQUESTED** [review]
This needs to be `keg_only` or install `base64` into `libexec`. Let's see what other maintainers have to say.

**Comment by SMillerDev** [comment]
Is say it's stale but needs to be keg only

File: Formula/libb64.rb
Comment (MikeMcQuaid) [inline Formula/libb64.rb]: Looks like a newer version: https://sourceforge.net/projects/libb64/files/libb64/libb64/libb64-1.2.1.zip/download

This hasn't been updated for a very long time but it's packaged by Debian (https://packages.debian.org/source/stretch/libb64)

File: Formula/libb64.rb
Comment (MikeMcQuaid) [reply Formula/libb64.rb]: > Should this be keg only if accepted?

Yes.

File: Formula/libb64.rb
Comment (MikeMcQuaid) [inline Formula/libb64.rb]: Note to other maintainers: there is no `make install`.

File: Formula/libb64.rb
Comment (MikeMcQuaid) [reply Formula/libb64.rb]: > This definitely needs a better test. Homebrew maintainers typically have their hands pretty full, so we'll probably need your help @schloegl to get this going.

Yes.




> For that purpose starting base64 seems to be a good enough test.

This starts it, does nothing and verifies an exit code. It'd be preferable to use `assert_match` and `shell_output` (and shouldn't take long to do).

Comment (schloegl) [reply Formula/libb64.rb]: > If you only need the library, you could write a small C program in the tetst (or similar) to ensure you can link against the library and/or install a simple test into the `pkg_share` directory during install and then compile & link in the test.


I really need only libb64.a.  Can you point me to some example  (e.g. in some other formula, or in the homebrew documentation) how such a test should look like ?

Comment (MikeMcQuaid) [reply Formula/libb64.rb]: We can have an additional test for the library if desired (see `libxml2.rb` for a library test) but this formula will need to have a test for the binary too and need it to be installed, sorry.

File: Formula/libb64.rb
Comment (MikeMcQuaid) [inline Formula/libb64.rb]:
```suggestion
```

File: Formula/libb64.rb
Comment (MikeMcQuaid) [inline Formula/libb64.rb]:
```suggestion
```

File: Formula/libb64.rb
Comment (MikeMcQuaid) [inline Formula/libb64.rb]:
```suggestion
    system ENV.cc, "test.c", "-lb64", "-o", "test"
```

File: Formula/libb64.rb
Comment (zbeekman) [reply Formula/libb64.rb]: @schloegl Can you please update to the most recent version?

File: Formula/libb64.rb
Comment (zbeekman) [inline Formula/libb64.rb]:
```suggestion
  keg_only :provided_by_macos

  def install
```

File: Formula/libb64.rb
Comment (zbeekman) [inline Formula/libb64.rb]:
```suggestion
  url "https://downloads.sourceforge.net/project/libb64/libb64/libb64/libb64-1.2.1.zi
p"
```

File: Formula/libb64.rb
Comment (zbeekman) [inline Formula/libb64.rb]:
```suggestion
  sha256 "20106f0ba95cfd9c35a13c71206643e3fb3e46512df3e2efb2fdbf87116314b2"
```

File: Formula/libb64.rb
Comment (zbeekman) [inline Formula/libb64.rb]: Could you validate that b64 encoding was correct? You could write the b64 encoding to stdout or a file in your test, and compare that to the expected output. Would be good to test the executable too, if you want this formula added.

**Review by zbeekman — DISMISSED** [review]
@schloegl Looks like you're making good progress! Can you please address Mike's comments on the test, as well as mine for updating the version? If all tests pass locally and on CI (including `brew audit --strict --online libb64`), after you've addressed the changes, it would be great if you could squash everything into one commit.

Thanks!

**Comment by schloegl** [comment]
> @schloegl Looks like you're making good progress! Can you please address Mike's comments on the test, as well as mine for updating the version? If all tests pass locally and on CI (including `brew audit --strict --online libb64`), after you've addressed the changes, it would be great if you could squash everything into one commit.
> 
> Thanks!



> @schloegl Looks like you're making good progress! Can you please address Mike's comments on the test, as well as mine for updating the version? If all tests pass locally and on CI (including `brew audit --strict --online libb64`), after you've addressed the changes, it would be great if you could squash everything into one commit.
> 
> Thanks!

I think I've addresses all requests, including the upgrade to 1.2.1. 

I failed in squashing the commits into a single commit(I tried the instructions from  https://gist.github.com/patik/b8a9dc5cd356f9f6f980 but it did not work). What's the procedure you recommend ? 
I'd setup a new branch and a new merge request, but I guess there is a better way.

File: Formula/libb64.rb
Comment (MikeMcQuaid) [inline Formula/libb64.rb]: Are you sure this is necessary? Try without it.

File: Formula/libb64.rb
Comment (MikeMcQuaid) [reply Formula/libb64.rb]: This can be skipped given the binary isn't installed.

File: Formula/libb64.rb
Comment (MikeMcQuaid) [reply Formula/libb64.rb]: I think this is fine as-is.

File: Formula/libb64.rb
Comment (MikeMcQuaid) [inline Formula/libb64.rb]:
```suggestion
```

File: Formula/libb64.rb
Comment (MikeMcQuaid) [inline Formula/libb64.rb]:
```suggestion
```

File: Formula/libb64.rb
Comment (MikeMcQuaid) [inline Formula/libb64.rb]:
```suggestion
```

**Comment by MikeMcQuaid** [comment]
@schloegl Please grant maintainers permission to push to your fork (checkbox on the right-hand side of the PR), thanks!

File: Formula/libb64.rb
Comment (schloegl) [reply Formula/libb64.rb]: > @schloegl Please grant maintainers permission to push to your fork (checkbox on the right-hand side of the PR), thanks!

That checkbox was already selected, you should be able to push.

**Comment by MikeMcQuaid** [comment]
Thanks so much for your first contribution! Without people like you submitting PRs we couldn't run this project. You rock, @schloegl!

**Pull request closed without merging**
